lib/web_ui/lib/src/engine/navigation/history.dart

```diff
@@ -100,7 +100,7 @@ abstract class BrowserHistory {
   ///
   /// Subclasses should send appropriate system messages to update the flutter
   /// applications accordingly.
-  void onPopState(covariant DomPopStateEvent event);
+  void onPopState(Object? state);
```

Comment:
I'm not sure I follow the reasoning for this change. Could you clarify?

Below, the override is still defined as `void Function(covariant DomPopStateEvent)`.

Reply:
For reasons that I don't really want to write a massive essay about right now (but would be happy to talk about more offline if you want) I don't want any of the surface area of this API to be JS interop objects. It gets complicated with `package:web` and some of the future stuff there.